src/processor.cpp

```diff
@@ -0,0 +1,164 @@
+// This file is a part of Julia. License is MIT: https://julialang.org/license
+
+// Processor feature detection
+
+#include "julia.h"
+#include "julia_internal.h"
+
+extern "C" {
+
+// CPUID
+
+#if defined(_CPU_X86_) || defined(_CPU_X86_64_)
+JL_DLLEXPORT void jl_cpuid(int32_t CPUInfo[4], int32_t InfoType)
+{
+#if defined _MSC_VER
+    __cpuid(CPUInfo, InfoType);
+#else
+    __asm__ __volatile__ (
+#if defined(__i386__) && defined(__PIC__)
+        "xchg %%ebx, %%esi;"
+        "cpuid;"
+        "xchg %%esi, %%ebx;":
+        "=S" (CPUInfo[1]) ,
+#else
+        "cpuid":
+        "=b" (CPUInfo[1]),
+#endif
+        "=a" (CPUInfo[0]),
+        "=c" (CPUInfo[2]),
+        "=d" (CPUInfo[3]) :
+        "a" (InfoType)
+        );
+#endif
+}
+
+JL_DLLEXPORT void jl_cpuidex(int32_t CPUInfo[4], int32_t InfoType, int32_t subInfoType)
+{
+#if defined _MSC_VER
+    __cpuidex(CPUInfo, InfoType, subInfoType);
+#else
+    __asm__ __volatile__ (
+#if defined(__i386__) && defined(__PIC__)
+        "xchg %%ebx, %%esi;"
+        "cpuid;"
+        "xchg %%esi, %%ebx;":
+        "=S" (CPUInfo[1]) ,
+#else
+        "cpuid":
+        "=b" (CPUInfo[1]),
+#endif
+        "=a" (CPUInfo[0]),
+        "=c" (CPUInfo[2]),
+        "=d" (CPUInfo[3]) :
+        "a" (InfoType),
+        "c" (subInfoType)
+        );
+#endif
+}
+#endif
+
+// -- set/clear the FZ/DAZ flags on x86 & x86-64 --
+#ifdef __SSE__
+
+// Cache of information recovered from jl_cpuid.
+// In a multithreaded environment, there will be races on subnormal_flags,
+// but they are harmless idempotent races.  If we ever embrace C11, then
```

Comment:
We do. And I've already replaced this with a global variable with dynamic initialization on my local version.